cli.go

```diff
@@ -132,6 +112,52 @@ func (h *CLIHandler) Auth(c *api.Client, m map[string]string) (*api.Secret, erro
 	}
 }
 
+func callbackHandler(c *api.Client, mount string, doneCh chan<- loginResp) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		var response string
+		var secret *api.Secret
+		var err error
+
+		defer func() {
+			w.Write([]byte(response))
+			doneCh <- loginResp{secret, err}
+		}()
+
+		// Pull any parameters from either the body or query parameters.
+		// FormValue prioritizes body values, if found.
+		data := map[string][]string{
+			"state":    {req.FormValue("state")},
+			"code":     {req.FormValue("code")},
+			"id_token": {req.FormValue("id_token")},
+		}
+
+		// If this is a POST, then the form_post response_mode is being used and the flow
+		// involves an extra step. First GET the data to Vault, and then issue a GET with
+		// the same state/code to complete the auth as normal.
+		if req.Method == http.MethodPost {
+			url := c.Address() + path.Join("/v1/auth", mount, "oidc/callback")
```

Comment:
There's no existing API Client logical functions that can submit this POST request, similar to how we use `ReadWithData` below?

Reply:
Most of the API Client functions choke if they don't get a secret back. There are a couple of "Raw" flavored ones, but I found them to be a pain since they're not set up for forms. So I have to manual encode everything, construct a body reader, set correct content type, etc. All the stuff `PostForm` is already handling.

Reply:
Okies